File: src/main/java/com/imagegame/service/UserPreferencesService.java
package com.imagegame.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.imagegame.domain.UserPreferences;
import com.imagegame.dto.UserPreferencesDTO;
import com.imagegame.dto.UserPreferencesSearchDTO;
import com.imagegame.dto.UserPreferencesPageDTO;
import com.imagegame.dto.UserPreferencesConvertCriteriaDTO;
import com.imagegame.service.GenericService;
import com.imagegame.dto.common.RequestDTO;
import com.imagegame.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface UserPreferencesService extends GenericService<UserPreferences, Integer> {

	List<UserPreferences> findAll();

	ResultDTO addUserPreferences(UserPreferencesDTO userPreferencesDTO, RequestDTO requestDTO);

	ResultDTO updateUserPreferences(UserPreferencesDTO userPreferencesDTO, RequestDTO requestDTO);

    Page<UserPreferences> getAllUserPreferencess(Pageable pageable);

    Page<UserPreferences> getAllUserPreferencess(Specification<UserPreferences> spec, Pageable pageable);

	ResponseEntity<UserPreferencesPageDTO> getUserPreferencess(UserPreferencesSearchDTO userPreferencesSearchDTO);
	
	List<UserPreferencesDTO> convertUserPreferencessToUserPreferencesDTOs(List<UserPreferences> userPreferencess, UserPreferencesConvertCriteriaDTO convertCriteria);

	UserPreferencesDTO getUserPreferencesDTOById(Integer userPreferencesId);







}

File: imagegame_ui/src/services/UserPreferencesService.js
import http from "../http-common"; 

class UserPreferencesService {
  getAllUserPreferencess(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/userPreferences/userPreferencess`, searchDTO);
  }

  get(userPreferencesId) {
    return this.getRequest(`/userPreferences/${userPreferencesId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/userPreferences?field=${matchData}`, null);
  }

  addUserPreferences(data) {
    return http.post("/userPreferences/addUserPreferences", data);
  }

  update(data) {
  	return http.post("/userPreferences/updateUserPreferences", data);
  }
  
  uploadImage(data,userPreferencesId) {
  	return http.postForm("/userPreferences/uploadImage/"+userPreferencesId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new UserPreferencesService();
